default/python/universe_generation/universe_generator.py

```diff
@@ -119,11 +122,52 @@ def create_universe(psd_map):
         raise Exception(err_msg)
     print("Home systems:", home_systems)
 
-    # set up empires for each player
+    teams = {}
+    for psd in psd_map.values():
+        if psd.starting_team >= 0:
+            teams[psd.starting_team] = teams.get(psd.starting_team, 0) + 1
+    teams = {k: v for k, v in teams.items() if v > 1}
+    print("Teams: ", teams)
+
     seed_rng(seed_pool.pop())
-    for empire, psd, home_system in zip(psd_map.keys(), psd_map.values(), home_systems):
-        if not setup_empire(empire, psd.empire_name, home_system, psd.starting_species, psd.player_name):
-            report_error("Python create_universe: couldn't set up empire for player %s" % psd.player_name)
+    if len(teams) > 0:
+        # choose team's core home system
+        home_system_teams = home_system_team_core(home_systems, sorted(teams.items(), key=itemgetter(1), reverse=True))
+        print("Home systems team core: ", home_system_teams)
+        # choose order of placing teams exclude already placed cores
+        placement_list = []
+        for team in home_system_teams.keys():
+            placement_list += [(i, team) for i in range(teams[team] - (1 if team in home_system_teams.keys() else 0))]
+        placement_list.sort(reverse=True)
+        placement_teams = [i[1] for i in placement_list]
+        print("Placement teams: ", placement_teams)
+        # calculate placement based on teams
+        layout = home_system_layout(home_systems, systems)
+        print("Home systems layout: ", layout)
+        psds = list(psd_map.items())
+        for home_system, team in place_teams(layout, home_system_teams, placement_teams):
+            home_systems.remove(home_system)
+            psds_new = list()
+            placed = False
+            for empire, psd in psds:
+                if placed or psd.starting_team != team:
+                    psds_new.append((empire, psd))
+                else:
+                    placed = True
+                    if not setup_empire(empire, psd.empire_name, home_system, psd.starting_species, psd.player_name):
+                        report_error("Python create_universe: couldn't set up empire for player %s" % psd.player_name)
+            if not placed:
+                report_error("Python create_universe: couldn't set up empire for team %d" % team)
+            psds = psds_new
+        # place leftovers
+        for empire, psd, home_system in zip(psds, home_systems):
+            if not setup_empire(empire, psd.empire_name, home_system, psd.starting_species, psd.player_name):
+                report_error("Python create_universe: couldn't set up empire for player %s" % psd.player_name)
```

Comment:
If it's not too complicated, I'd prefer if most of that part (lines 134-165) could also be moved to a function and into `teams.py` or `empires.py`.

Reply:
I'll look into it.

Reply:
Moved that part.